packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -1387,6 +1399,29 @@ export default class TradeStore extends BaseStore {
             has_only_forward_starting_contracts =
                 ContractType.getContractCategories().has_only_forward_starting_contracts;
         }
+
+        // Set stake to default one (from contracts_for) on symbol or contract type switch.
+        // On contract type we also additionally reset take profit
+        if (this.default_stake && this.is_dtrader_v2_enabled) {
```

Comment:
What is happening:
In `Stake` component we are setting `default_stake` to _trade-store_ via `setDefaultStake`. `default_stake` - is BE default stake value, extracted from `contracts_for` API in `useContractsForCompany` hook.
Whenever user is changing `contract_type` or `symbol`, `processNewValuesAsync` function is called. Inside it, we do some necessary checks (like if it's dtrader-v2 and etc) and set `default_stake` (and `TP`) as a proper fields in proposal request object.

Reply:
Why here, why not is `Stake` component: less `proposal` calls, less rerenders, no need in `usePrevious` hook (it will be needed in case if we store this logic in `Stake`) as in store we have access to both old and to new values.

Also, `Stake` component is already huge and horrible.